src/Features/Core/Portable/GenerateMember/GenerateVariable/AbstractGenerateVariableService.cs

```diff
@@ -216,10 +216,23 @@ private static void GenerateWriteableField(ArrayBuilder<CodeAction> result, Sema
                 if (containingMethod.Parameters.Length > 0)
                 {
                     var compilation = await document.Project.GetRequiredCompilationAsync(cancellationToken).ConfigureAwait(false);
-                    var lastParameterIsCancellationToken = Equals(containingMethod.Parameters.Last().Type, compilation.CancellationTokenType());
 
-                    if (lastParameterIsCancellationToken)
-                        parameterIndex--;
+                    for (var i = containingMethod.Parameters.Length - 1; i >= 0; i--)
+                    {
+                        var parameter = containingMethod.Parameters[i];
+                        if ((parameter.HasExplicitDefaultValue
+                            || Equals(parameter.Type, compilation.CancellationTokenType())
```

Comment:
💡 Consider extracting this value so it doesn't have to call `CancellationTokenType()` repeatedly